src/unity/toolkits/supervised_learning/supervised_learning.cpp

```diff
@@ -1050,9 +1052,6 @@ void supervised_learning_model_base::api_train(
     
     valid_y = validation_data.select_columns({target}).materialize_to_sframe();
 
-    
-    check_target_column_type(this->name(), valid_y);
```

Comment:
Should we no longer be checking the validation set too?

Reply:
Validation set is created from the dataset that is provided by the user. So I feel it might be repetitive to also call the check function again on a section of that dataset.

Reply:
Although validation data can be automatically sampled from the training data, I believe our API also allows users to provide the validation set as a completely unrelated SFrame.

Reply:
Yes, you're right. Thank you for pointing that! It was worth verifying that case as well. So, in the case when a user provides their own validation set: the model `create()` calls `_validate_data(dataset, validation_set)`. The validation_set is set to 'auto' if we want the validation set to the generated automatically or it can be given an SFrame containing the validation set. In the latter case, `_validate_data(dataset, validation_set)` applies append on the two SFrames using `validation_set.head().append(dataset.head())`  to check schema. This allows verifying that the columns and their types of both `dataset` and `validation_data` match.
Since we are now checking the type of the `dataset` target column, and by extension we make sure that the `validation_data` also has the matching valid column type.

Reply:
I think the check should probably be in both places.  That's the easiest fix here.

Reply:
Good idea, to be safe the check on validation data column type should be restored.